### PR TITLE
WFCORE-932 stop-embedded-server stops the embedded host controller and vice versa

### PR DESCRIPTION
**Pull request by luck3y**
Restrict which server stop commands are available based on the type started.

### CODE REVIEW

File: cli/src/main/java/org/jboss/as/cli/embedded/StopEmbeddedHostControllerHandler.java [hunk 1]
```diff
@@ -52,7 +52,7 @@ public boolean isAvailable(CommandContext ctx) {
     @Override
     protected void doHandle(CommandContext ctx) throws CommandLineException {
         EmbeddedServerLaunch hostControllerLaunch = hostControllerReference.get();
-        if (hostControllerLaunch != null) {
+        if (hostControllerLaunch != null && hostControllerLaunch.getHostController() != null) {
```
Comment (bstansberry) [inline cli/src/main/java/org/jboss/as/cli/embedded/StopEmbeddedHostControllerHandler.java]: Shouldn't this be in isAvailable()? That's what will prevent the CLI offering the option.

Having it here too is fine, but people shouldn't see the command if it's not valid.

Comment (luck3y) [inline cli/src/main/java/org/jboss/as/cli/embedded/StopEmbeddedHostControllerHandler.java]: This is calling EmbeddedServerLaunch.getHostController() which in turn looks at EmbeddedServerReference to see if its a running hostcontroller or a standalone server. If this is a standalone server here, then we just ignore the shutdown, as standalone handles it instead.

As I recall it was a lot cleaner to implement it this way from the CLI side, but the issue was that you could use either shutdown command to shutdown either embedded process type.

**Comment by luck3y** [comment]
retest this please

**Comment by luck3y** [comment]
retest this please

**Comment by bstansberry** [comment]
@pferraro the SingletonDeploymentDescriptorTestCase failure shown in one of the CI comments above seems to be failing pretty regularly. Twice more here: https://github.com/wildfly/wildfly-core/pull/1112

**Pull request merged**